extensions/python/pythonprocessors/nifiapi/properties.py

```diff
@@ -281,3 +284,21 @@ def getProperty(self, descriptor) -> PythonPropertyValue:
             expression_language_support = descriptor.expressionLanguageScope != ExpressionLanguageScope.NONE
         property_value = self.cpp_context.getProperty(property_name)
         return PythonPropertyValue(self.cpp_context, property_name, property_value, expression_language_support)
+
+    def getStateManager(self) -> StateManager:
+        return self.cpp_context.getStateManager()
+
+    def getName(self) -> str:
+        return self.cpp_context.getName()
+
+    def getProperties(self) -> Dict[PropertyDescriptor, str]:
+        properties = dict()
+        cpp_properties = self.cpp_context.getProperties()
+
+        for property_name in cpp_properties:
+            for property_descriptor in self.processor.getPropertyDescriptors():
+                if property_descriptor.name == property_name:
+                    properties[property_descriptor] = cpp_properties[property_name]
+                    break
```

Comment:
This is quadratic, we should improve this later to have linear complexity. It's fine for now, but I made a Jira issue to fix this later: [MINIFICPP-2361](https://issues.apache.org/jira/browse/MINIFICPP-2361)

Reply:
can't we just flip it and do, `cpp_properties` seems to be a wrapped python dictionary, forwarding the `contains` call (which I think the following code would do)
```suggestion
        for property_descriptor in self.processor.getPropertyDescriptors():
            if property_descriptor.name in cpp_properties:
                properties[property_descriptor] = cpp_properties[property_name]
```

Reply:
I updated the test and the implementation of getProperties in 9eec8f9b0a61264578330919ed9e16fa2c5007fd

Reply:
That's a hash table lookup now, right? That would be amortized linear complexity, which is fine IMO. If you can confirm, I'll close the Jira issue.

Reply:
yes, I think the Jira issue can be closed now